pkg/build/api/v1/types.go

```diff
@@ -371,6 +375,13 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty" description:"forces the source build to pull the image if true"`
 }
 
+// BuildPostCommitSpec holds a build post commit hook specification.
+type BuildPostCommitSpec struct {
+	// RunBash holds a bash script to be executed in a container running the
+	// build output image.
+	RunBash string `json:"runBash,omitempty" description:"bash script to be executed in a container running the build output image"`
```

Comment:
We don't need to override entrypoint in the API, because the code that invokes this should bypass entrypoint.  We do not want this command path to use entrypoint unless we can make a really clear justification that entrypoint fits in the use case (I haven't heard that yet)

I think Docker has the right idea here - I want this to execute in a shell-like context most of the time, I want to use working dir most of the time, and entrypoint of the run image is not the target 99% of the time.  I don't think that means that we do the automatic "/bin/bash -c" unless we find that the use case does not work without it.  If I specify an absolute path, Bash really shouldn't be involved.  A user should be able to use env vars from the container definition in their call (behaving exactly like upstream Command) to avoid that need.